handlers/graph.go

```diff
@@ -172,15 +172,15 @@ func buildNamespaceTrees(namespace string, o options.Options, client *prometheus
 	seenNodes := make(map[string]*tree.ServiceNode)
 
 	// Query for root nodes. Root nodes must originate outside of the requested
-	// namespace (typically "unknown").  Destination nodes must be in the namespace.
+	// namespace.  Destination nodes must be in the namespace.
 	namespacePattern := fmt.Sprintf(".*\\\\.%v\\\\..*", namespace)
 	query := fmt.Sprintf("sum(rate(%v{source_service!~\"%v\",destination_service=~\"%v\",response_code=~\"%v\"} [%vs])) by (%v)",
 		o.Metric,
-		namespacePattern,          // regex for namespace-constrained service
-		namespacePattern,          // regex for namespace-constrained service
-		"[2345][0-9][0-9]",        // regex for valid response_codes
-		int(o.Duration.Seconds()), // range duration for the query
-		"source_service")          // group by
+		namespacePattern,                // regex for namespace-constrained service
+		namespacePattern,                // regex for namespace-constrained service
+		"[2345][0-9][0-9]",              // regex for valid response_codes
+		int(o.Duration.Seconds()),       // range duration for the query
+		"source_service,source_version") // group by
```

Comment:
it doesn't looks like it's related to this PR purpose, is it something that was accidentally missing ? (i'd be fine with that, just in order to understand)

Reply:
You are correct, I removed the assumption that the version would be 'unknown'.  This came up while I was manually testing this PR.